File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/classes/abstractClass/Dog.java
package de.fraunhofer.iem.classesAndObjects.classes.abstractClass;

/**
 * Class that extends Animal
 *
 * @author Ranjith Krishnamurthy
 */
public class Dog extends Animal {
    public Dog(String DNA) {
        super(DNA);
    }

    @Override
    public void makeNoise() {
        System.out.println("Bow Bow");
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/classes/abstractClass/Dog.kt
package de.fraunhofer.iem.classesAndObjects.classes.abstractClass

/**
 * Class that extends Animal
 *
 * @author Ranjith Krishnamurthy
 */
class Dog(DNA: String) : Animal(DNA) {
    override fun makeNoise() {
        println("Bow Bow")
    }
}